micro-rdk/src/common/config_monitor.rs

```diff
@@ -71,16 +77,47 @@ where
                 })
                 .await?;
 
-            if new_config
-                .config
-                .is_some_and(|cfg| cfg != *self.curr_config)
-            {
-                if let Err(e) = self.storage.reset_robot_configuration() {
-                    log::warn!(
-                        "Failed to reset robot config after new config detected: {}",
-                        e
-                    );
-                } else {
+            if let Some(config) = new_config.as_ref().config.as_ref() {
+                let mut reboot = false;
+
+                #[cfg(feature = "ota")]
+                {
+                    if let Some(service) = config
+                        .services
+                        .iter()
+                        .find(|&service| service.model == *ota::OTA_MODEL_TRIPLET)
+                    {
+                        match ota::OtaService::from_config(
+                            service,
+                            self.storage.clone(),
+                            self.executor.clone(),
+                        ) {
+                            Ok(mut ota) => match ota.update().await {
+                                Ok(needs_reboot) => reboot = needs_reboot,
+                                Err(e) => log::error!("failed to complete ota update: {}", e),
+                            },
+                            Err(e) => log::error!(
+                                "failed to create ota service from config:{} - {:?}",
+                                e,
+                                service,
+                            ),
+                        }
+                    }
+                }
+
+                if *config != *self.curr_config {
+                    if let Err(e) = self.storage.reset_robot_configuration() {
+                        log::warn!(
+                            "Failed to reset robot config after new config detected: {}",
+                            e
+                        );
+                    } else {
+                        reboot = true;
+                    }
+                }
+
+                if reboot {
+                    // TODO(RSDK-9464): flush logs to app.viam before restarting
```

Comment:
This ticket would be another reason to DRY the shutdown handling.

Reply:
as in, have a global shutdown function that handles log-flushing?